extensions/common/vault/vault-hashicorp/src/test/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClientIntegrationTest.java

```diff
@@ -0,0 +1,200 @@
+/*
+ *  Copyright (c) 2023 Mercedes-Benz Tech Innovation GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Mercedes-Benz Tech Innovation GmbH - Implement automatic Hashicorp Vault token renewal
+ *
+ */
+
+package org.eclipse.edc.vault.hashicorp;
+
+import jakarta.json.Json;
+import org.eclipse.edc.junit.annotations.ComponentTest;
+import org.eclipse.edc.junit.extensions.EdcExtension;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.vault.VaultContainer;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.edc.vault.hashicorp.HashicorpVaultConfig.VAULT_TOKEN;
+import static org.eclipse.edc.vault.hashicorp.HashicorpVaultConfig.VAULT_TOKEN_RENEW_BUFFER;
+import static org.eclipse.edc.vault.hashicorp.HashicorpVaultConfig.VAULT_TOKEN_SCHEDULED_RENEWAL_ENABLED;
+import static org.eclipse.edc.vault.hashicorp.HashicorpVaultConfig.VAULT_TOKEN_TTL;
+import static org.eclipse.edc.vault.hashicorp.HashicorpVaultConfig.VAULT_URL;
+import static org.testcontainers.shaded.org.awaitility.Awaitility.await;
+
+@ComponentTest
+@Testcontainers
+@ExtendWith(EdcExtension.class)
+class HashicorpVaultClientIntegrationTest {
+    private static final String DOCKER_IMAGE_NAME = "vault:1.9.6";
+    private static final String ROOT_TOKEN = UUID.randomUUID().toString();
+    private static final long CREATION_TTL = 6L;
+    private static final long TTL = 5L;
+    private static final long RENEW_BUFFER = 4L;
+    private static final String CLIENT_TOKEN_KEY = "client_token";
+    private static final String AUTH_KEY = "auth";
+
+    @Container
+    static final VaultContainer<?> VAULTCONTAINER = new VaultContainer<>(DOCKER_IMAGE_NAME)
+            .withVaultToken(ROOT_TOKEN);
+
+    @BeforeEach
+    void beforeEach(EdcExtension extension) throws IOException, InterruptedException {
+        assertThat(CREATION_TTL).isGreaterThan(TTL);
+        extension.setConfiguration(getConfig());
+    }
+
+    @Test
+    void lookUpToken_whenTokenNotExpired_shouldSucceed(HashicorpVaultClient hashicorpVaultClient) {
+        var tokenLookUpResult = hashicorpVaultClient.lookUpToken();
+
+        assertThat(tokenLookUpResult.succeeded()).isTrue();
```

Comment:
The number of assertions here is a code smell. Please see my comments about using POJOs to deserialize a small number of properties. Related to this, integration tests shoould not be used to cover what can be done in unit tests.